homeassistant/components/rainbird/quality_scale.yaml

```diff
@@ -0,0 +1,85 @@
+rules:
+  # Bronze
+  config-flow:
+    status: todo
+    comment: Some fields are missing data descriptions.
+  brands: done
+  dependency-transparency: done
+  common-modules: done
+  has-entity-name: done
+  action-setup:
+    status: done
+    comment: |
+      The integration only has an entity service, registered in the platform.
+  appropriate-polling:
+    status: done
+    comment: |
+      Rainbird devices are local. Irrigation valve/controller status is polled
+      once per minute to get fast updates when turning on/off the valves.
+      The irrigation schedule uses a 15 minute poll interval since it rarely
+      changes.
+
+      Rainbird devices can only accept a single http connection, so this uses a
+      an aiohttp.ClientSession with a connection limit, and also uses a request
+      debouncer.
+  test-before-configure: done
+  entity-event-setup:
+    status: exempt
+    comment: Integration is polling and does not subscribe to events.
+  unique-config-entry: done
```

Comment:
My impression is that reconfigure flow is a new feature that was added that integrations need to add code to support and this is something that needs to be addressed when reconfigure support is added to the integration. Am I understanding that right?

Reply:
Integrations need to support this, but that means we should not have 2 ways for users to reconfigure an integration

Reply:
I'm not sure I understand.   Perhaps the missing context here is you can't change the host during reauth?

https://github.com/home-assistant/core/blob/502a221feb345ce434e265be5dcfb44176828950/homeassistant/components/rainbird/config_flow.py#L42

Reply:
Yes! And reauth in theory has to wait before something raises ConfigEntryAuthFailed, while this way users can proactively fix it (for example just after they changed the password or something)

Reply:
I'm a bit lost here. I'm not sure if you are misunderstanding the current state of how this works or if i'm misunderstanding your point.

Can you be more explicit in (1) what you're looking for here and spell out in more detail or (2) what you think the current code is doing? 

I don't undertand what "Integrations need to support this, but that means we should not have 2 ways for users to reconfigure an integration".

Here is the current state as I understand ig:
- the integration supports reauth today, accepting only a password. Note the schema is different. This is not reconfiguration, and does not allow what reconfiguration allows.
- the integration does not support reconfiguration *yet* in any way.  This should happen when as it raises up the quality scale ladder.

My impression is that reconfiguration does not invalidate existing reauth flows. (Otherwise, it sounds half baked)

Reply:
What i understood the current code was doing was allowing to start the user flow again and setup the same device and have it update the current information. This is because the unique id is set and then we abort if it's configured and update the information passed to it.

(Although because you push back on it, I'm sure that I probably missed some check somewhere) 

If this logic is indeed implemented like this, we should move that to a reconfigure flow since we should not update current entries via the user flow

Reply:
OK i may not fully understand these APIs. https://github.com/home-assistant/core/pull/101603#discussion_r1390187273 has some discussion about when this was added -- and https://github.com/home-assistant/core/pull/101603#discussion_r1390187405 

I will add a reconfigure flow and we can revisit all this?

Also, if you have suggestions for a test case to add to exercise a specific behavior I can add them and we can see what happens. I think there are a lot of existing test cases, but not sure if they are hitting the specific scenario.